### PR TITLE
Address feedback

### PR DESCRIPTION
**Pull request by pmorie**

### CODE REVIEW

File: docs/design/secrets.md [hunk 1]
```diff
@@ -184,4 +190,4 @@ container requiring the secret. This makes it vulnerable to attacks on the conta
 especially if the secret is placed in plain text. MCS labels can mitigate some of this risk.
 However if there is a particular use case for a very sensitive secret, the secret itself could be
 stored encrypted and placed in encrypted form in the file system for the container. The container
-would have to know how to decrypt it.
+would have to know how to decrypt it and would receive the decryption key via another channel.
```
Comment (EricMountain-1A) [inline docs/design/secrets.md]: If a node is compromised, then all containers it handles are compromised too: e.g. if you own the node, you can make it "docker exec" into any container. If a container is compromised, its secrets are also compromised.

Any secret handled by a container is only going to be as safe as the whole stack.  To obtain a decryption key via another channel in a secure+authenticated manner, it needs to prove it has the right to that key.  To achieve that, it needs to hold a secret...  So the sensitive secret will only be as safe as the secrets handed to the container.  If "very sensitive secrets" are required in the sense that this proposal is considered insufficient, then this software stack is not "secure enough" and you probably want an HSM-based solution or something like that.

In short, I'd get rid of the part starting "However...".

File: docs/design/secrets.md [hunk 2]
```diff
@@ -81,8 +86,8 @@ Collecting remaining TODOs here:
 1.  Describe the lifecycle of secrets
 2.  Determine how/whether secrets interact with service accounts; are namespaces enough for the
     security scope? Can/should my security context affect what secrets I have access to?
-3.  How will pods know when secret values change?  Should there be a way to express that a pod
-    which consumes a secret should be restarted when the secret changes?
+3.  How will containers know when secret values change?  Should there be a way to express that a
```
Comment (EricMountain-1A) [inline docs/design/secrets.md]: I'm inclined to say that a running container is not made aware of a change in secrets.

Comment (pmorie) [inline docs/design/secrets.md]: I agree that secrets should be immutable from the standpoint of a running container; I'll add some more context to this so that's clear.
